service/frontend/workflowHandler.go

```diff
@@ -761,6 +788,57 @@ func (wh *WorkflowHandler) PollForDecisionTask(
 	return resp, nil
 }
 
+func (wh *WorkflowHandler) getIsolationGroup(ctx context.Context, domainName string) string {
+	if wh.config.EnableTasklistIsolation(domainName) {
+		return partition.IsolationGroupFromContext(ctx)
+	}
+	return ""
+}
+
+func (wh *WorkflowHandler) getPartitionConfig(ctx context.Context, domainName string) map[string]string {
+	if wh.config.EnableTasklistIsolation(domainName) {
+		return partition.ConfigFromContext(ctx)
+	}
+	return nil
+}
+
+func (wh *WorkflowHandler) isIsolationGroupHealthy(ctx context.Context, domainName, isolationGroup string) bool {
+	if wh.GetIsolationGroupState() != nil && wh.config.EnableTasklistIsolation(domainName) {
+		isDrained, err := wh.GetIsolationGroupState().IsDrained(ctx, domainName, isolationGroup)
+		if err != nil {
+			wh.GetLogger().Error("Failed to check if an isolation group is drained, assume it's healthy", tag.Error(err))
+			return true
+		}
+		return !isDrained
+	}
+	return true
+}
+
+func (wh *WorkflowHandler) waitUntilIsolationGroupHealthy(ctx context.Context, domainName, isolationGroup string) bool {
```

Comment:
To me, this is super clear, but to anyone without context this would be super confusing as to it's purpose. Do you mind adding a comment describing the problem of creating a retry-storm and blocking